hack/lib/update_olm_manifest.sh

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+filename="olm-bindata-"$1.go
+echo $filename
+
+function get_olm_manifests() {
+    mkdir olm-manifests
+    echo "downloading olm manifests for version ${1}"
+    wget -O olm-manifests/olm.yaml "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/${1}/olm.yaml"
+    wget -O olm-manifests/crds.yaml "https://github.com/operator-framework/operator-lifecycle-manager/releases/download/${1}/crds.yaml"
+}
+
+function remove_olm_manifests {
+    rm -rf olm-manifests
+}
+
+# check for files starting with the name "olm-bindata" inside internal/olm folder
+function delete_old_olmbindata {
+    echo "Deleting previous versions of olm-bindata files if they exist"
+    find internal/olm -maxdepth 1 -type f -name olm-bindata-* -exec rm {} \;
+}
+
+FILE=internal/olm/$filename
+if [ ! -f "$FILE" ]; then
+    delete_old_olmbindata
+    get_olm_manifests $1
+
+    go get -u github.com/go-bindata/go-bindata/...
+    $(go env GOPATH)/bin/go-bindata -o olm-bindata-$1.go -pkg olm olm-manifests/
+    mv olm-bindata-$1.go internal/olm/
+
+    remove_olm_manifests
+fi
+
+go mod tidy
```

Comment:
It is not clear for me. Why  we are adding a binary in  the /olm/internal ? https://github.com/operator-framework/operator-sdk/tree/ae4a1002a5096b4b91c485757555a9470f2f4285/internal/olm

<img width="793" alt="Screenshot 2020-07-08 at 09 29 44" src="https://user-images.githubusercontent.com/7708031/86896224-9490d500-c0fd-11ea-9088-5895cbd03c5c.png">

- We should not commit binaries
- All bins should be built in the /build/bin instead of that. 

Also, should not the OLM manifests be shipped with SDK binary instead of we have another one? How it will works? If we do another bin then users need to download this bin as well.

Reply:
This isn't a "binary" per se, it is [bindata](https://godoc.org/github.com/jteeuwen/go-bindata). The reasoning here is we want to avoid network calls and default to manifests we know work when possible, so each SDK release embeds a set of OLM manifests to use as defaults to `operator-sdk olm` subcommands.

This file should live elsewhere, ex. `bindata/olm-manifests.go`.

Reply:
so, the files should be got and used in the release. Why we should commit that? IMO the reason for do not commit binary are the same and are applied at this case as well.

Reply:
Committing bindata solves several problems for users and is more efficient than the current solution.

Reply:
Hi @estroz, 

IMO we should not commit the bindata into the repo.  See that is not a good practice to commit something that is built/generated. 

See that we need is to provide the manifests when we run the `olm install`. And then, the solution is to ship it with SdK binary. Am I right? So, we need to have the data inside of SDK binary which also is not committed. We can generate the bin add do all it is required in the release process and we run `make install` but we do need to commit it. 

If we would commit something that it would need to be the manifests files that are used to generate the bindata instead of it. 

See for example the Kubernetes repo working with https://github.com/kubernetes/kubernetes/search?q=bindata&unscoped_q=bindata

They are using the build dir or the test dir to output the bindata when it is required but the file is never committed for the same reasons. 

c/c @jmrodri  




We are getting the OLM manifests and building the bindata to ship it inside of the SDK binary. 


Am I right? It is not useful outside of it and we should not commit the bindata in the same way that we do not commit the SDK binary. 

If we cannot rely on OLM and expected that they can remove the manifests that are used to gen the bindata then, we should copy and commit in the repo the **manifest files** and NOT the bindata which is generated using them. At the same way that we do not commit the SDK binary and just the code that used to gen it. (which IMO is not the case btw since OLM is part of the org)

See for example the Kubernetes repo working with https://github.com/kubernetes/kubernetes/search?q=bindata&unscoped_q=bindata

They are using the build dir or the test dir to output the bindata when it is required but the file is never committed for the same reasons. 

c/c @jmrodri

Reply:
Hi @estroz, 

IMO we should NOT commit the bindata into the repo.  See that is not a good practice to commit something that is built/generated. Also, note that we need to provide the manifests when we run the `olm install`. And then, the solution is to ship it with SDK binary. So, we need to have the data inside of SDK binary which also is not committed. in this way, we can generate the bindata when is required if we have the files(release process and make install) but we do need to commit it. 

So, If we would commit something then, it would be the manifests files that are used to generate the bindata instead of it. (however, I do not think that it is required since we get it from OLM which is in the same org)

See for example the Kubernetes repo working with https://github.com/kubernetes/kubernetes/search?q=bindata&unscoped_q=bindata. They are using the build dir or the test dir to output the bindata when it is required but the file is never committed for the same reasons. 

just to clarifies, my suggestion here is: 
- Solve the problem as it has been solved  (get the manifest, gen the bindata and add it in the SDK bin) when we run make install and for the release process
- Generate the bindata in the build dir and not in operator-sdk/internal/olm
- Not commit the bindata 
- Add in the .gitigore file a new entry for we ensure that it will not be committed by mistake

@jmrodri @fabianvf @joelanford wdyt?  




We are getting the OLM manifests and building the bindata to ship it inside of the SDK binary. 


Am I right? It is not useful outside of it and we should not commit the bindata in the same way that we do not commit the SDK binary. 

If we cannot rely on OLM and expected that they can remove the manifests that are used to gen the bindata then, we should copy and commit in the repo the **manifest files** and NOT the bindata which is generated using them. At the same way that we do not commit the SDK binary and just the code that used to gen it. (which IMO is not the case btw since OLM is part of the org)

See for example the Kubernetes repo working with https://github.com/kubernetes/kubernetes/search?q=bindata&unscoped_q=bindata

They are using the build dir or the test dir to output the bindata when it is required but the file is never committed for the same reasons. 

c/c @jmrodri

Reply:
Hi @estroz, 

IMO we should NOT commit the bindata into the repo.  See that is not a good practice to commit something that is built/generated. Also, note that we need to provide the manifests when we run the `olm install`. And then, the solution is to ship it with SDK binary. So, we need to have the data inside of SDK binary which also is not committed. in this way, we can generate the bindata when is required if we have the files(release process and make install) but we do need to commit it. 

So, If we would commit something then, it would be the manifests files that are used to generate the bindata instead of it. (however, I do not think that it is required since we get it from OLM which is in the same org)

See for example the Kubernetes repo working with https://github.com/kubernetes/kubernetes/search?q=bindata&unscoped_q=bindata. They are using the build dir or the test dir to output the bindata when it is required but the file is never committed for the same reasons. 

just to clarifies, my suggestion here is: 
- Solve the problem as it has been solved  (get the manifest, gen the bindata and add it in the SDK bin) when we run make install and for the release process
- Generate the bindata in the build dir and not in operator-sdk/internal/olm
- Not commit the bindata 
- Add in the .gitigore file a new entry for we ensure that it will not be committed by mistake

@jmrodri @fabianvf @joelanford wdyt?

Reply:
I think it is solved 👍